app/workers/atmosphere/vm_template_monitoring_worker.rb

```diff
@@ -36,13 +36,20 @@ def update_images(tenant, images)
       all_tenant_templates = tenant.virtual_machine_templates.to_a
       images.each do |image|
         updated_vmt = Cloud::VmtUpdater.new(tenant, image).execute
-
         all_tenant_templates.delete(updated_vmt)
       end
 
       #remove deleted templates
       all_tenant_templates.each do |vmt|
-        vmt.destroy(false) if vmt.old?
+        # Unlink VMT from current tenant first
+        tenant.virtual_machine_templates = tenant.virtual_machine_templates-[vmt]
```

Comment:
Yep, good catch :clap: 
I have also determined that it's not necessary to invoke `tenant.save` and `vmt.reload`.